thirdparty/eigen3/CMakeLists.txt

```diff
@@ -6,6 +8,7 @@ set(Eigen3_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/eigen3-prefix/src/eigen3)
 
 ExternalProject_Add(
     eigen3
+    SYSTEM
```

Comment:
I faintly remember that the reason for ExternalProject here might have been to suppress warnings. Could be worth a try to see if FetchContent works now?